openwisp_monitoring/device/base/models.py

```diff
@@ -183,6 +183,15 @@ def _transform_data(self):
                 for signal_key, signal_values in interface['mobile']['signal'].items():
                     for key, value in signal_values.items():
                         signal_values[key] = float(value)
+            # If HT/VHT is not being used, set the HT/VHT
+            # field of WiFi clients to None. This is necessary
+            # because some clients may be VHT capable but VHT is
+            # not enabled at the radio level, which can mislead
+            # into thinking the client is not HT/VHT capable.
+            if 'wireless' in interface and 'clients' in interface['wireless']:
+                for client in interface['wireless']['clients']:
+                    client['ht'] = client['ht'] if client['ht'] else None
+                    client['vht'] = client['vht'] if client['vht'] else None
```

Comment:
This code doesn't look right.
Are you sure this code implements what is requested in the issue description?

> Depends on https://github.com/openwisp/openwisp-monitoring/issues/350.
> 
> If not using HT/VHT, flag the HT/VHT field of WiFi clients as null and show it as with the unkonwn gray question mark icon instead of the red x icon.
> 
> This is because if, for example, VHT is not used, then all the clients will have the VHT flag set as False, which may not be true, some clients may be VHT capable but VHT is not enabled at the radio level, hence is shown as false, but it's misleading.

It's being asked to show "None" if the interface is not using either HT or VHT.

Reply:
Sorry, I misunderstood the issue description. Therefore, I believe we need to update this implementation to use the radio config `htmode`. If that flag is set to `NOHT`, it means that `HT` and `VHT` are disabled at the radio level. In that case, we should set the `HT` and `VHT` fields to `None`, which would result in showing an unknown icon on the admin.